brave/src/main/java/brave/propagation/ExtraField.java

```diff
@@ -0,0 +1,331 @@
+   * underlying field.
+   */
+  public static void setValue(TraceContext context, String name, String value) {
+    PropagationFields.put(context, validateName(name), value, ExtraFieldPropagation.Extra.class);
+  }
+
+  /**
+   * Sets the current value of the field with the specified key, or drops if not a configured
+   * field.
+   *
+   * <p>Prefer using {@link ExtraField#setValue(String)} when you have a reference to the
+   * underlying field.
+   * <p>Prefer {@link #setValue(TraceContext, String, String)} if you have a reference to the trace
+   * context.
+   *
+   * @see ExtraField#name()
+   */
+  public static void setValue(String name, String value) {
+    TraceContext context = currentTraceContext();
+    if (context != null) setValue(context, name, value);
+  }
+
+  /** @since 5.11 */
+  public static class Builder {
+    final String name;
+    String prefix;
+    boolean redacted;
+
+    Builder(String name) {
+      this.name = validateName(name);
+    }
+
+    /**
+     * @see #prefix()
+     * @since 5.11
+     */
+    public Builder prefix(String prefix) {
+      if (prefix == null) throw new NullPointerException("prefix == null");
+      if (prefix.isEmpty()) throw new IllegalArgumentException("prefix is empty");
+      this.prefix = prefix;
+      return this;
+    }
+
+    /**
+     * @see #redacted()
+     * @since 5.11
+     */
+    public Builder redacted() {
+      this.redacted = true;
+      return this;
+    }
+
+    /**
+     * Ensures this field can integrate with {@link CorrelationFieldScopeDecorator}.
+     *
+     * @since 5.11
+     */
+    public CorrelationBuilder withCorrelation() {
+      return new CorrelationBuilder(this);
+    }
+
+    /** @since 5.11 */
+    public ExtraField build() {
+      return new ExtraField(this);
+    }
+  }
+
+  /** Used to make an extra field for {@link CorrelationFieldScopeDecorator} */
+  public static class CorrelationBuilder {
+    final Builder delegate;
+
+    CorrelationBuilder(Builder delegate) {
+      this.delegate = delegate;
+    }
+
+    /** @see Builder#prefix(String) */
+    public CorrelationBuilder prefix(String prefix) {
+      delegate.prefix(prefix);
+      return this;
+    }
+
+    /** @see Builder#redacted() */
+    public CorrelationBuilder redacted() {
+      delegate.redacted();
+      return this;
+    }
+
+    /** @see Builder#build() */
+    public final WithCorrelation build() {
+      return new WithCorrelation(delegate);
+    }
+  }
+
+  public static final class WithCorrelation extends ExtraField implements CorrelationField {
+    WithCorrelation(Builder builder) {
+      super(builder);
+    }
+  }
+
+  final String prefix;
+  final String name;
+  final boolean redacted;
+
+  ExtraField(Builder builder) {
+    prefix = builder.prefix;
+    name = builder.name;
+    redacted = builder.redacted;
+  }
+
+  /** Extracts the value for this field in the context or null if unavailable. */
+  @Nullable public String getValue(TraceContext context) {
```

Comment:
I guess these can delegate to the static methods instead of copy-paste

Reply:
this drifted after we now use the field type internally.